src/Microsoft.TemplateEngine.Edge/Settings/ComponentManager.cs

```diff
@@ -19,25 +19,46 @@ internal class ComponentManager : IComponentManager
         private readonly Dictionary<Type, HashSet<Guid>> _componentIdsByType;
         private readonly SettingsStore _settings;
         private readonly ISettingsLoader _loader;
+        private readonly ComponentCache _componentCache = new ComponentCache();
 
-        private interface ICache
+        private class ComponentCache 
```

Comment:
Should we just drop this class, and add `_parts` dictionary to `ComponentManager` and same goes for both methods...